drivers/can/sja1000.h

```diff
@@ -0,0 +1,911 @@
+/****************************************************************************
+ * drivers/can/sja1000.h
+ *
+ * SJA1000 CAN driver based on esp32c3_twai.h
+ *
+ * License header retained from original source.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __DRIVERS_CAN_SJA1000_H
+#define __DRIVERS_CAN_SJA1000_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Default interrupts - Exclude data overrun (bit[3]) and brp_div (bit[4])
+ */
+
+#define SJA1000_DEFAULT_INTERRUPTS 0xe7
+
+/* Number of standard/extended filters */
+
+#define SJA1000_NFILTERS           1
+
+/* SJA1000_MODE_REG register
+ * Mode Register
+ */
+
+#define SJA1000_MODE_REG(baseaddr) (baseaddr + 0x0)
```

Comment:
I do not like this style to provide base into macros.  It has been simplification for the case where chip has only single CAN (TWAI) controller, i.e. the base in macros has been fixed
```
#define DR_REG_TWAI_BASE                        0x3ff6b000
```
It is error prone and problematic when base is configurable. There has to be at least one additional rackets in the macro around `baseaddr`. But even that would not work correctly for different registers span, see later comment